sqlitecluster/SQLite.cpp

```diff
@@ -296,7 +296,15 @@ SQLite::~SQLite() {
 
 bool SQLite::beginTransaction(TRANSACTION_TYPE type) {
     if (type == TRANSACTION_TYPE::EXCLUSIVE) {
-        _sharedData.commitLock.lock();
+        if (isSyncThread) {
+            // Blocking the sync thread has catastrophic results (forking) and so we either get this quickly, or we fail the transaction.
+            if (!_sharedData.commitLock.try_lock_for(5s)) {
+                SWARN("Failed to acquire commit lock in sync thread exclusive transaction!");
```

Comment:
If the sync thread fails to get a lock in five seconds, just fail and move on. The command will need to be resent, but at least we aren't keeping the whole cluster from working.